code/datums/components/mood.dm

```diff
@@ -301,5 +302,38 @@
 		if(ETHEREAL_CHARGE_ALMOSTFULL to ETHEREAL_CHARGE_FULL)
 			add_event(null, "charge", /datum/mood_event/charged)
 
+
+/datum/component/mood/proc/HandleHygiene(mob/living/carbon/human/H)
+	switch(H.hygiene)
+		if(0 to HYGIENE_LEVEL_DIRTY)
+			if(has_trait(TRAIT_NEAT))
+				add_event(null, "neat", /datum/mood_event/dirty)
+			if(has_trait(TRAIT_NEET))
```

Comment:
stupid ass trait system, shoudl be datums so you can register signals to the trait to generate the mood events

Reply:
yell at the traits guy

Reply:
The traits guy is gone.

Reply:
i said traits not quirks